conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/bitcoin/BitcoinOperations.scala

```diff
@@ -0,0 +1,89 @@
+package tech.cryptonomic.conseil.indexer.bitcoin
+
+import cats.effect.{Concurrent, Resource}
+import com.typesafe.scalalogging.LazyLogging
+import fs2.Stream
+import slickeffect.Transactor
+
+import tech.cryptonomic.conseil.common.config.Platforms.BitcoinBatchFetchConfiguration
+import tech.cryptonomic.conseil.common.rpc.RpcClient
+import tech.cryptonomic.conseil.common.bitcoin.BitcoinPersistence
+import tech.cryptonomic.conseil.common.bitcoin.rpc.BitcoinClient
+import tech.cryptonomic.conseil.indexer.config.{Custom, Depth, Everything, Newest}
+
+/**
+  * Bitcoin operations for Lorre.
+  *
+  * @param bitcoinClient Bitcoin client instance
+  * @param persistence Bitcoin persistence instance
+  * @param tx [[slickeffect.Transactor]] to perform a Slick operations on the database
+  */
+class BitcoinOperations[F[_]: Concurrent](
+    bitcoinClient: BitcoinClient[F],
+    persistence: BitcoinPersistence[F],
+    tx: Transactor[F],
+    batchConf: BitcoinBatchFetchConfiguration
+) extends LazyLogging {
+
+  /**
+    * Start Lorre with mode defined with [[Depth]].
+    *
+    * @param depth Can be: Newest, Everything or Custom
+    */
+  def loadBlocks(depth: Depth): Stream[F, Unit] =
+    Stream
+      .eval(tx.transact(persistence.getLatestIndexedBlock))
+      .zip(bitcoinClient.getBlockChainInfo)
+      .evalTap(_ => Concurrent[F].delay(logger.info(s"Start Lorre for Bitcoin")))
+      .flatMap {
+        case (block, info) =>
+          depth match {
+            case Newest => loadBlocksWithTransactions(block.map(_.height).getOrElse(1) to info.blocks)
+            case Everything => loadBlocksWithTransactions(1 to info.blocks)
+            case Custom(depth) => loadBlocksWithTransactions((info.blocks - depth) to info.blocks)
+          }
+      }
+
+  /**
+    * Get Blocks from Bitcoin node through Bitcoin client and save them into the database using Slick.
+    * In the beginning, the current list of blocks is obtained from the database and removed from the computation.
+    *
+    * @param range Inclusive range of the block's height
+    */
+  def loadBlocksWithTransactions(range: Range.Inclusive): Stream[F, Unit] =
+    Stream
+      .eval(tx.transact(persistence.getExistingBlocks(range)))
+      .flatMap(
+        existingBlocks =>
+          Stream
+            .range(range.start, range.end)
+            .filter(height => !existingBlocks.contains(height))
+            .through(bitcoinClient.getBlockHash(batchConf.hashBatchSize))
+            .through(bitcoinClient.getBlockByHash(batchConf.blocksBatchSize))
+            .through(bitcoinClient.getBlockWithTransactions(batchConf.transactionsBatchSize))
+            .evalTap { case (block, _) => Concurrent[F].delay(logger.debug(s"Save block with height: ${block.height}")) }
```

Comment:
What about logging every X block? I believe we talked about that. Current solution will create to much noice and we can miss something.

Reply:
This is a debug log message. It isn't the default behavior

Reply:
Ah, you are right! Question, do we have any feedback from Lorre, while blocks are indexed? I'd like to avoid empty console, while process is working properly, maybe we could still have some feedback, but not that frequent like with every saved block?

Reply:
ok :) I will add something like that

Reply:
done